docs/framework/react/guide/file-based-routing.md

```diff
@@ -329,6 +348,26 @@ export default defineConfig({
 })
 ```
 
+### Ignoring the generated route tree file
+
+If your project is configured to use a linter and/or formatter, you may want to ignore the generated route tree file. This is this file is managed by TanStack Router and shouldn't and therefore shouldn't be changed by your linter or formatter.
+
+Here are some resources to help you ignore the generated route tree file:
+
+- Prettier - [https://prettier.io/docs/en/ignore.html#ignoring-files-prettierignore](https://prettier.io/docs/en/ignore.html#ignoring-files-prettierignore)
+- ESLint - [https://eslint.org/docs/latest/use/configure/ignore#ignoring-files](https://eslint.org/docs/latest/use/configure/ignore#ignoring-files)
+- Biome - [https://biomejs.dev/reference/configuration/#filesignore](https://biomejs.dev/reference/configuration/#filesignore)
+
+If you are using VSCode, you can also add the following to your `.vscode/settings.json` file to exclude the generated route tree file from the editor's file watcher:
+
+```json
+{
+  "files.watcherExclude": {
+    "**/routeTree.gen.ts": true
+  }
+}
+```
+
```

Comment:
Will this cause problems with intellisense not picking up changes to the file made by the generator? In my quick and dirty testing it doesn't seem to be the case. But when running neovim, I need to make sure that my editor loads modifications to the generated route tree for the language server to be able to know about them.

Regardless, is there even a real benefit to ignoring it from the file watcher? I don't think VSCode would do anything itself to the file when notified of a change, and having language intelligence services be aware of changes as soon as possible is probably a good thing.

Reply:
This won't have any bearing on the intellisense service provided by VSCode which bases the majority of its analysis based on what is returned by `tsc`. As such, this change would have no bearing on it.

Currently, when VSCode's background file watcher gets notified of an update to the active/inactive/closed generated route tree file (triggered by actions like adding or removing a route), it then proceeds to open the file in your active workbench in its _previous_ state, thereby interrupting the save triggered by the router-generator. This sometimes leads to the file being formatted incorrectly or having syntactical errors. This is a long-standing issue we've had and this change solves it.